sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -235,6 +239,21 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         "write files of Hive data source directly.")
     }
 
+    val hadoopConf = df.sparkSession.sessionState.newHadoopConf()
+    val defaultFS = FileSystem.getDefaultUri(hadoopConf).getScheme
+    val tmpFS = new URI(System.getProperty("java.io.tmpdir")).getScheme
+
+    val isTempCheckpointLocationAvailable = tmpFS match {
+      case null | "file" =>
+        if (defaultFS == null || defaultFS.equals("file")) {
+          true
+        } else {
+          false
+        }
+      case defaultFS => true
```

Comment:
So @mgaido91 based on the logics here, if `tmpFs` and `defaultFs` are both HDFS, this will return `true`, right?